mesa/time.py

```diff
@@ -278,8 +278,13 @@ def step_type(self, type_class: Type[Agent], shuffle_agents: bool = True) -> Non
         for agent_key in agent_keys:
             self.agents_by_type[type_class][agent_key].step()
 
-    def get_type_count(self, type_class: Type[Agent]) -> int:
+    def get_type_count(self, type_class: Type[Agent], filter_func: Callable[[Agent], bool] = None) -> int:
```

Comment:
This is too much of a corner case to warrant putting it in the library? I think it is better for the example to define a new scheduler that inherits from `RandomActivationByType` and overrides this method instead.